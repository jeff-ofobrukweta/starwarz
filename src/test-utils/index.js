import { mount } from "enzyme";
import { Provider } from "react-redux";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import logger from "redux-logger";
import thunk from "redux-thunk";
import rootReducer from "../redux/rootReducers";

export const testStore = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, logger))
);

export const reduxify = (Component, props = {}, state = {}) => {
  return function reduxWrap() {
    return (
      <Provider store={testStore}>
          <Component {...props} />
      </Provider>
    );
  };
};


export const makeMountRender = (Component, defaultProps = {}) => {
  return (customProps = {}) => {
    const props = {
      ...defaultProps,
      ...customProps
    };
    return mount(<Component {...props} />);
  };
};


export const snapshotify = reactWrapper => {
  return reactWrapper.html();
};