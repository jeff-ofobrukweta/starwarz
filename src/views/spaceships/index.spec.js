import { makeMountRender, reduxify, snapshotify } from "../../test-utils";
import Spaceship from "../spaceships/index";


describe("Fetching Actions from Starwars", () => {
  it("Snapshot Spaceships Layout", () => {
    const wrapper = makeMountRender(reduxify(Spaceship))()
    expect(snapshotify(wrapper)).toMatchSnapshot();
  });
});