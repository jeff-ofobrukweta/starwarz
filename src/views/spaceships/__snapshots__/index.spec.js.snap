// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetching Actions from Starwars Snapshot Spaceships Layout 1`] = `"<div class=\\"films-page\\"><div class=\\"film-details\\"><div class=\\"details-wrapper\\"><div class=\\"loading-characters-section loading-skelecton\\"><div class=\\"loader-card-main-inner-container\\"><div class=\\"list-icon-card-main\\"><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div></div><div class=\\"child-main-contain\\"><div class=\\"holder-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main\\"></div></div></div></div><div class=\\"loading-characters-section loading-skelecton\\"><div class=\\"loader-card-main-inner-container\\"><div class=\\"list-icon-card-main\\"><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div></div><div class=\\"child-main-contain\\"><div class=\\"holder-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main\\"></div></div></div></div><div class=\\"loading-characters-section loading-skelecton\\"><div class=\\"loader-card-main-inner-container\\"><div class=\\"list-icon-card-main\\"><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div></div><div class=\\"child-main-contain\\"><div class=\\"holder-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main\\"></div></div></div></div><div class=\\"loading-characters-section loading-skelecton\\"><div class=\\"loader-card-main-inner-container\\"><div class=\\"list-icon-card-main\\"><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div></div><div class=\\"child-main-contain\\"><div class=\\"holder-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main\\"></div></div></div></div><div class=\\"loading-characters-section loading-skelecton\\"><div class=\\"loader-card-main-inner-container\\"><div class=\\"list-icon-card-main\\"><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div></div><div class=\\"child-main-contain\\"><div class=\\"holder-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main\\"></div></div></div></div><div class=\\"loading-characters-section loading-skelecton\\"><div class=\\"loader-card-main-inner-container\\"><div class=\\"list-icon-card-main\\"><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div><div class=\\"icon-container-wrapper icon-holder-loading-skelecton\\"></div></div><div class=\\"child-main-contain\\"><div class=\\"holder-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main icon-holder-loading-skelecton\\"></div><div class=\\"holder-sub-title-main\\"></div></div></div></div></div></div></div>"`;
